### PR TITLE
[WIP] New ops wrapper

### PR DESCRIPTION
**Pull request by alexanderst**
## What changes were proposed in this pull request?

1. New ops wrappers in nd4j.
2. Bugfix in NLP.

## How was this patch tested?
Unit tests.

### CODE REVIEW

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class AdjustContrast extends BaseAdjustContrast {
+
+    public AdjustContrast() {super();}
+
+    public AdjustContrast(INDArray in, double factor, INDArray out) {
+        super(in, factor, out);
+    }
+
+    @Override
+    public String opName() {
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java:22]: You should also define tensorflow opname. There's method for that.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    public AdjustContrastV2() {super();}
+
+    public AdjustContrastV2(INDArray in, double factor, INDArray out) {
+        super(in, factor, out);
+    }
+
+    @Override
+    public String opName() {
+        return "adjust_contrast_v2";
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java:23]: Same as above

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.nd4j.base.Preconditions;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+
+public abstract class BaseAdjustContrast extends DynamicCustomOp {
+    public BaseAdjustContrast() {
+    }
+
+    public BaseAdjustContrast(INDArray in, double factor, INDArray out) {
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java:13]: you also need SDVariable-based constructors here, as other ops have

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    public BitCast(INDArray in, int dataType, INDArray out) {
+        inputArguments.add(in);
+        outputArguments.add(out);
+        iArguments.add(Long.valueOf(dataType));
+    }
+
+    @Override
+    public String opName() {
+        return "bitcast";
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java:25]: same as above

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import org.nd4j.linalg.api.buffer.DataType;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+import org.nd4j.linalg.factory.Nd4j;
+
+public class BitCast extends DynamicCustomOp {
+    public BitCast() {}
+
+    public BitCast(INDArray in, int dataType, INDArray out) {
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java:13]: same as above

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package org.nd4j.linalg.api.ops.custom;
+
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.api.ops.DynamicCustomOp;
+import org.nd4j.linalg.factory.Nd4j;
+
+public class CompareAndBitpack extends DynamicCustomOp {
+    public CompareAndBitpack() {}
+
+    public CompareAndBitpack(INDArray in, double threshold, INDArray out) {
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java:12]: Same as above

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        inputArguments.add(in);
+        inputArguments.add(Nd4j.scalar(threshold));
+        outputArguments.add(out);
+    }
+
+    @Override
+    public String opName() {
+        return "compare_and_bitpack";
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java:24]: Same as above

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/DivideNoNan.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+    public DivideNoNan(INDArray in1, INDArray in2, INDArray out) {
+        inputArguments.add(in1);
+        inputArguments.add(in2);
+        outputArguments.add(out);
+    }
+
+    @Override
+    public String opName() {
+        return "divide_no_nan";
```
Comment (raver119) [inline nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/DivideNoNan.java:25]: Same as above

**Review by raver119 — CHANGES_REQUESTED** [review]
Generally LGTM, but there's no SameDiff/TF import integration yet.

So you should also add these wrappers to DifferentiableFunctionFactory and corresponding namespaces

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java (hunk 1 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java:22]: Added.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java (hunk 2 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java:23]: Fixed.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java (hunk 3 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java:13]: Added.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java (hunk 1 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrast.java:22]: Done.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java (hunk 2 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/AdjustContrastV2.java:23]: Done

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java (hunk 3 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BaseAdjustContrast.java:13]: Done.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java (hunk 4 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java:25]: Done

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java (hunk 5 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/BitCast.java:13]: Done.

File: nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java (hunk 6 above)
Comment (alexanderst) [reply nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/custom/CompareAndBitpack.java:12]: Done.

**Review by raver119 — APPROVED** [review]

**Pull request merged**